src/user/profile/profile-feed.js

```diff
@@ -0,0 +1,251 @@
+import React, { Component } from "react";
+import "./profile-feed.scss";
+import {
+  FaComments,
+  FaGlobe,
+  FaStickyNote,
+  FaCalendar,
+} from "react-icons/fa";
+import { connect } from "react-redux";
+import {
+  getProfile,
+  getEventsCreatedByUser,
+  getProjectCreatedByUser,
+  getPostsCreatedByUser,
+} from "../../actions/usersAction";
+
+import { resetComments } from "../../actions/commentAction";
+import { Card, Image } from "react-bootstrap";
+import { Circle } from "react-shapes";
+import ReadMe from "./ReadMe";
+import userIcon2 from "../../images/userIcon2.jpg";
+import parse from "html-react-parser";
+import PostContent from "./content/PostContent";
+
+class ProfileFeed extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: "All",
+      userProfile: {},
+      userEvents: [],
+      userProjects: [],
+      userPosts: [],
+      displayPostContent: false,
+      displayingPost: 0,
+    };
+  }
+  componentDidMount() {
+    console.log(this.props);
+    setTimeout(() => {
+      this.props.getProfile();
+    });
+    setTimeout(() => {
+      this.props.getPostsCreatedByUser();
+    });
+    setTimeout(() => {
+      this.props.getEventsCreatedByUser();
+    });
+    setTimeout(() => {
+      this.props.getProjectCreatedByUser();
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillRecieveProps` has been deprecated / marked unsafe and its use should be avoided. Maybe we could use `getDerivedStateFromProps` or something else to do the same thing. Here is a link to official docs - https://reactjs.org/docs/react-component.html

Reply:
@ksraj123  This should be discussed on today's meeting as ComponentWillReceiveProps is used throughout the application at the moment.